README.md

```diff
@@ -42,8 +19,230 @@ RedHat modifications against the upstream Che include:
+```
+
+This script:
+- runs the upstream Che maven build on the openshift connector related modules
+- updates the RH distribution by running RH maven build with the options passed
+as arguments,
+- generates the docker images for the upstream and the RH distribution 
+and tag them appropriately.
+
+The available parameters are the same as for the `build_fabric8.sh` script.
+
+### Examples
+
+- To clean/rebuild *only* the Bayesian language server, without any validation
+check or test, use the following command:
+
+```bash
+dev-scripts/build_fabric8_and_deploy.sh -Dfast -Dpl="plugins/che-plugin-bayesian-lang-server" -Damd clean
+```
+
+- If your upstream Che repository is on the "myFixBranch" git branch,
+to rebuild the upstream openshift connector integration, and have the
+RedHat distribution updated accordingly, use the following command:
+
+```bash
+dev-scripts/update_openshift_connector.sh -Dfast -Dche.branch="myFixBranch" clean
+```
+
+### Docker scripts to create and tag images from the RH build artifacts 
+
+##### In the local docker environment
+
+```bash
+dev-scripts/docker_create_images_and_tag.sh
+```
+__Note:__ This step is already included in the build scripts.
+However, if you plan to deploy or rollupdate to Minishift, you should also create / tag docker images
+in the Minishift docker environment, as detailed in next section. 
+
+##### In the Minishift docker environment
+
+```bash
+dev-scripts/docker_create_images_and_tag_in_minishift.sh
+```
+
+## How to deploy / roll-update it in Minishift
+    
+### Deployment prerequisites
+
+* Get [minishift](https://github.com/minishift/minishift#installation) (we have tested with v1.0.0beta4)
+  * Minishift v1.0.0beta4 includes the `oc` binary, and this should be added to your `$PATH`. The binary is located
+    in the `.minishift/cache/oc/<oc version>/` directory (e.g. on Linux, it is in `~/.minishift/cache/oc/v1.4.1/`).
+* Get [gofabric8](https://github.com/fabric8io/gofabric8#getting-started) (we have tested with v0.4.121)
+* Clone fabric8-online git repository:
+
+`git clone https://github.com/fabric8io/fabric8-online.git`
+
+* The scripts assume that the `fabric8-online` repository is cloned in `${HOME}/github/fabric8-online/`.
+If you cloned it in another location, you need to override the `FABRIC8_ONLINE_PATH` variable
+with the full path of the cloned `fabric8-online` git repository :
+
+```bash
+export FABRIC8_ONLINE_PATH="<path where you cloned it>/fabric8-online/"
+```
+__Important note:__ the `FABRIC8_ONLINE_PATH` should always end with a `/`.
+
+* For all the scripts that deploy or rollout to Minishift, you should have minishift running.
+This can be done with the following command:
+
+```bash
+minishift start
+```
+
+#### Deploy to Minishift 
+
+__Important__: Before deploying to Minishift, ensure that you created / tagged the Che docker images 
+*in the Minishift docker environment* (see [previous section](#in-the-minishift-docker-environment))
```

Comment:
I'm not sure that it's completely true. Users that just want to deploy to minishift (without building Che first) don't need to create/tag Che Docker images. I think that will be the majority of people that are going to read this section. So I would move this sentence as a warning a few lines below.